packages/@aws-cdk/aws-stepfunctions/test/state-machine.test.ts

```diff
@@ -278,6 +280,75 @@ describe('State Machine', () => {
     });
   });
 
+  test('Instantiate a State Machine with a task assuming a literal roleArn', () => {
```

Comment:
Can you also add a test using `fromRoleName` using a stack with a different
account from the state machine? I want to make sure it builds the arn/reference
correctly.

Reply:
@humanzz I think this is still unresolved.

Reply:
I added a new test case, but now rereading your message, I'm not 100% am doing what you wanted.

Do you want a test case with 2 stacks, one having the role, the other having the state machine, both stacks having different `Environment` so that they each have a different account Id?